index.js

```diff
@@ -1,68 +1,52 @@
 // import express (after npm install express)
-const express = require('express');
+const express = require('express')
 
 // create new express app and save it as "app"
-const app = express();
+const app = express()
 
-//import package.json
-const pack = require('./package');
+// import package.json
+const packageInfo = require('./package')
 
-//import categories.json
-const all_categories = require('./src/data/categories.json').categories;
-
-//import products.json
-const all_products = require('./src/data/products').products;
+// import dataService.js
+const dataService = require('./dataService.js')
 
 // server configuration
-const PORT = 8080;
+const PORT = 8080
 
 // create a route for the app
 app.get('/', (req, res) => {
-  res.send('Hello World');
-});
-
-function construct_master(){
-	var master_products = all_products;
-	master_products.forEach(
-		(product) => product.categoryName = all_categories.filter(
-		// check if id of category element matches product categoryId
-			each_category => each_category.id === product.categoryId)[0].categoryName);
-	return master_products;
-}
-
-function get_product_by_category(ctyId){ 
-    var category_products =  product_master.filter(
-    //filter out product which have specific categoryId
-    product => product.categoryId === ctyId);
-    return(category_products);
-}
-
-var product_master = construct_master();
+  res.send('Hello World')
+})
 
 // endpoint to get all products
 app.get('/products/all', (req, res) => {
-  res.json(product_master);
-});
+  let allProducts = {}
+  for ([id, product] of dataService.getCombinedProductMap()) { allProducts[id] = product }
+  res.json(allProducts)
+})
 
 // endpoint to get details of specific product
 app.get('/product/:id', (req, res) => {
-  var requested_product = product_master.find(
-    //filter out product which have specific id
-    product => product.id === req.params.id);
-  res.json(requested_product);
-});
+  res.json(dataService.getCombinedProductMap().get(req.params.id))
```

Comment:
```suggestion
  res.json(dataService.getCombinedProductMap()[req.params.id])
```
Better to do a null check also